python-threatexchange/threatexchange/cli/dataset_cmd.py

```diff
@@ -255,3 +279,38 @@ def execute_generate_indices(self, settings: CLISettings) -> None:
             index = index_cls.build(signals.items())
             settings.index.store(s_type, index)
             self.stderr(f"Index for {s_type.get_name()} ready")
+
+    def _print_stdout(
+        self,
+        collab_name: str,
+        signal_type: SignalType,
+        signal_str: str,
+        metadata: FetchedSignalMetadata,
+    ) -> None:
+        if not self.signals_only and len(self.only_collabs) != 1:
+            print(repr(collab_name), end=" ")
+        if len(self.only_signals) != 1:
+            print(signal_type.get_name(), end=" ")
+        print(signal_str, end="")
+        if not self.signals_only:
+            print("", metadata, end="")
+        print()
+
+    def _print_csv(
+        self,
+        csvwriter: csv.DictWriter,
+        collab_name: str,
+        signal_type: SignalType,
+        signal_str: str,
+        metadata: FetchedSignalMetadata,
+    ) -> None:
+        agg = metadata.get_as_aggregate_opinion()
```

Comment:
blocking: I think this means using `signals_only` and `csv` will result in error.

Reply:
Ah, that's why I had that argument listed as optional originally.